packages/waas/src/secure-store.ts

```diff
@@ -0,0 +1,59 @@
+import { openDB } from 'idb'
+
+export interface SecureStoreBackend {
+    get(dbName: string, dbStoreName: string, key: string): Promise<string | null>
+    set(dbName: string, dbStoreName: string, key: string, value: string): Promise<boolean>
+    delete(dbName: string, dbStoreName: string, key: string): Promise<boolean>
+}
+
+export const getDefaultSecureStoreBackend = (): SecureStoreBackend | null => {
+    if (isIndexedDbAvailable()) {
+        return new IndexedDbSecureStoreBackend()
+    } else {
+        return null
+    }
+}
+
+export function isIndexedDbAvailable(): boolean {
+    return typeof indexedDB === 'object'
+}
+
+export class IndexedDbSecureStoreBackend implements SecureStoreBackend {
+    constructor() {
+        if (!isIndexedDbAvailable()) {
+            throw new Error('IndexedDB is not available')
+        }
+    }
+
+    async get(dbName: string, dbStoreName: string, key: string): Promise<string | null> {
+        const db = await openDB(dbName)
+        const tx = db.transaction(dbStoreName, 'readonly')
+        const value = await db.get(dbStoreName, key)
+        await tx.done
+        db.close()
+        return value
+    }
+
+    async set(dbName: string, dbStoreName: string, key: string, value: string): Promise<boolean> {
+        const db = await openDB(dbName, 1, {
```

Comment:
lets move this to the open() method too. I think its safe to call upgrade() each time on open, the idea is it will create a new db and set the version to 1 if its never been created before according to https://github.com/jakearchibald/idb

Reply:
Agreed – Previous implementation was only making this call when setting the key, but I think it's better to just do it on any open automatically. I'll change this.